crates/core/tedge_mapper/src/c8y/dynamic_discovery.rs

```diff
@@ -19,81 +22,58 @@ pub struct DiscoverOp {
-            DynamicDiscoverOpsError::FailedtoAddWatch(ops_dir.to_string_lossy().to_string())
-        })?;
-    Ok(inotify)
-}
+/// depending on which editor you use, temporary files could be created.
+/// this `operation_name_is_valid` fn will ensure that only files created
+/// that start with `c8y_` and contain alphabetic chars are allowed.
+fn operation_name_is_valid(operation: &str) -> bool {
+    let c8y_index = 4;
+    let (prefix, clipped_operation) = operation.split_at(c8y_index);
 
-pub fn create_inofity_event_stream(
-    ops_dir: PathBuf,
-) -> Result<inotify::EventStream<[u8; 1024]>, DynamicDiscoverOpsError> {
-    let buffer = [0; 1024];
-    let mut ino = create_inotify_watch(ops_dir)?;
-    Ok(ino.event_stream(buffer)?)
+    if prefix.eq(C8Y_PREFIX) {
+        clipped_operation.chars().all(|c| c.is_ascii_alphabetic())
+    } else {
+        false
+    }
 }
 
 pub fn process_inotify_events(
-    ops_dir: PathBuf,
-    event: Event<OsString>,
+    path: &Path,
+    mask: Masks,
 ) -> Result<Option<DiscoverOp>, DynamicDiscoverOpsError> {
-    if let Some(ops_name) = event.clone().name {
-        let operation_name = ops_name
-            .to_str()
-            .ok_or_else(|| DynamicDiscoverOpsError::NotAnOperationName(ops_name.clone()));
+    let operation_name = path
+        .file_name()
+        .and_then(|file_name| file_name.to_str())
+        .ok_or_else(|| DynamicDiscoverOpsError::NotAnOperation(path.to_path_buf()))?;
+
+    let parent_dir = path
+        .parent()
+        .ok_or_else(|| DynamicDiscoverOpsError::NotAnOperation(path.to_path_buf()))?;
 
-        match operation_name {
-            Ok(ops_name) => match event.mask {
-                EventMask::DELETE => {
-                    return Ok(Some(DiscoverOp {
-                        ops_dir,
-                        event_type: EventType::Remove,
-                        operation_name: ops_name.to_string(),
-                    }))
-                }
-                EventMask::CLOSE_WRITE => {
-                    return Ok(Some(DiscoverOp {
-                        ops_dir,
-                        event_type: EventType::Add,
-                        operation_name: ops_name.to_string(),
-                    }))
-                }
-                _ => return Ok(None),
-            },
-            Err(e) => return Err(e),
+    if operation_name_is_valid(operation_name) {
+        match mask {
+            Masks::Deleted => Ok(Some(DiscoverOp {
+                ops_dir: parent_dir.to_path_buf(),
+                event_type: EventType::Remove,
+                operation_name: operation_name.to_string(),
+            })),
+            Masks::Created => Ok(Some(DiscoverOp {
+                ops_dir: parent_dir.to_path_buf(),
+                event_type: EventType::Add,
+                operation_name: operation_name.to_string(),
+            })),
+            mask => {
+                warn!("Did nothing for mask: {}", mask);
+                Ok(None)
```

Comment:
Here is why I do think one needs to filter out all the unwatched events and remove the `Masks::Undefined` case. The user of the `tedge_utils::fs_notify` module will have nothing better to do.